src/plugins/intel_npu/tools/protopipe/src/simulation/layers_reader.cpp

```diff
@@ -42,5 +60,9 @@ InOutLayers LayersReader::readLayers(const InferenceParams& params) {
     // NB: Using OpenVINO to read the i/o layers information for *.onnx model
     OpenVINOParams ov;
     ov.path = OpenVINOParams::ModelPath{ort.model_path, ""};
-    return getOVReader().readLayers(ov, true /* use_results_names */);
+    auto inOutLayers = getOVReader().readLayers(ov, true /* use_results_names */);
+    adjustDynamicDims(inOutLayers.in_layers);
+    adjustDynamicDims(inOutLayers.out_layers);
```

Comment:
It would be strange to see `data will be provided as input` if you call this function on `output` layers

Reply:
Function no longer called for output layers. I was never sure it was required

Reply:
But seems it does. If output layer has `-1` it will be impossible to allocate memory for this buffer

Reply:
With or without updating output layer I saw no difference in the result.

Reply:
Simply because your model don't have dynamic shapes for outputs.